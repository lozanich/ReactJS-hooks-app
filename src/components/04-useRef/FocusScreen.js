import React, { useRef } from "react";
import "./focus.css";

export const FocusScreen = () => {
  const inputRef = useRef();

  const handleClick = () => {
    inputRef.current.select();
  };

  return (
    <div>
      <h1>Focus Screen</h1>
      <hr />

      <input ref={inputRef} className="form-control" placeholder="nombre" />

      <button onClick={handleClick} className="btn btn-outline-primary mt-5">
        Focus
      </button>
    </div>
  );
};
