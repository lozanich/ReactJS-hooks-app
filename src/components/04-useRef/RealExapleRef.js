import React, { useState } from "react";
import { MultipleCustomHooks } from "../examples/MultipleCustomHooks";
import "./focus.css";

export const RealExapleRef = () => {
  const [show, setShow] = useState(false);

  return (
    <div>
      <h1>Real Example useRef</h1>
      <hr />
      {show && <MultipleCustomHooks />}

      <button
        onClick={() => {
          setShow(!show);
        }}
        className="btn btn-primary mt-5"
      >
        show / hide
      </button>
    </div>
  );
};
