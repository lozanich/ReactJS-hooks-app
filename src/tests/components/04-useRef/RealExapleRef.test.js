import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { RealExapleRef } from "../../../components/04-useRef/RealExapleRef";

describe("Pruebas en el componente <RealExapleRef />", () => {
  test("Debe de coincidir con el snapshot", () => {
    const wrapper = shallow(<RealExapleRef />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostar el componente multiple custom hooks", () => {
    const wrapper = shallow(<RealExapleRef />);
    const button = wrapper.find("button");
    button.simulate("click");
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
  });
});
