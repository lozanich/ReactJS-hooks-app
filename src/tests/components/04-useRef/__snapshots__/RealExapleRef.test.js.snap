// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <RealExapleRef /> Debe de coincidir con el snapshot 1`] = `
<div>
  <h1>
    Real Example useRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    show / hide
  </button>
</div>
`;
